api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto

```diff
@@ -104,6 +104,12 @@ message HttpProtocolOptions {
     // via HTTP/3, and if the HTTP/3 connection fails, TCP (HTTP/1 or HTTP/2
     // based on ALPN) will be used instead.
     config.core.v3.Http3ProtocolOptions http3_protocol_options = 3;
+
+    // [#not-implemented-hide:]
+    // If HTTP/3 is present and an alternate protocol cache is present, attempts to connect
+    // will be via HTTP/3 only if the alternate protocol cache specifies an HTTP/3
```

Comment:
I'm having a hard time parsing this comment. What are you trying to say?

Reply:
I agree that the comment here is hard to digest.
IIUC this field is not specific to HTTP/3, but the comment doesn't describe what's its  purpose, and how it impacts the other protocols.
Also, not an expert here, but can the same cache in this case be for both HTTP/1.1 and HTTP/2 (or is it a different cache with a different configuration)?
If the cache isn't shared across the protocols, would it make sense to have this message inside each of the different Protocol-Options?

Reply:
How about having the comment focus on the main logic, and only noting the caveat. For example: `The presence of alternate protocols cache options causes the use of the alternate protocols cache, which is responsible for parsing and caching HTTP Alt-Svc headers. This enables the use of HTTP/3 for origins that advertise supporting it. Note that currently when the alternate protocols cache is absent we will always attempt HTTP/3 if HTTP/3 is enabled, but that behavior will be removed soon.`

Reply:
That sounds reasonable to me

Reply:
Let's doc up what the actual plan is, since it's hidden so the APIs won't be turned into docs. 
You can TODO here or in code to enforce the bits which aren't done yet.

Reply:
Sounds good to me! Done! (I hope :>) I took David's language and tweaked it slightly.